core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java

```diff
@@ -235,9 +244,11 @@ MainClassBuildItem build(List<StaticBytecodeRecorderBuildItem> staticInitTasks,
                 tryBlock.load(LaunchMode.DEVELOPMENT.equals(launchMode.getLaunchMode())));
 
         cb = tryBlock.addCatch(Throwable.class);
-        cb.invokeVirtualMethod(ofMethod(Throwable.class, "printStackTrace", void.class), cb.getCaughtException());
+        cb.invokeVirtualMethod(ofMethod(Logger.class, "error", void.class, Object.class, Throwable.class),
+                cb.readStaticField(logField.getFieldDescriptor()), cb.load("Failed to start application"),
+                cb.getCaughtException());
         cb.invokeVirtualMethod(ofMethod(StartupContext.class, "close", void.class), startupContext);
-        cb.throwException(RuntimeException.class, "Failed to start quarkus", cb.getCaughtException());
+        cb.throwException(RuntimeException.class, "Failed to start quarkus");
```

Comment:
There are test failures related to this change.

From what I saw in the log, the exception wasn't logged twice so I would keep the cause here.

Reply:
Hm, I tested my PR with quickstart and it was logged twice when I tried to run the app using the `-runner.jar`. I can try the dev mode and native image too..

Reply:
Actually some ArC tests failed because of this ;-).

Reply:
So I verified that it's logged twice but I agree that we should keep the cause here... I will update the PR shortly.